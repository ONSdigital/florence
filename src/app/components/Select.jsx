import React, {Component} from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    contents: PropTypes.array,
    label: PropTypes.string,
    id: PropTypes.string,
    override: PropTypes.bool,
    overrideLabel: PropTypes.string,
    overrideId: PropTypes.string,
    onChange: PropTypes.func
};

class Select extends Component {
    constructor(props) {
        super(props);
        this.handleChange = this.handleChange.bind(this);
        this.state = {
            value: ''
        }
    }

    handleChange(event) {
        this.setState({
            value: event.target.value
        });
        this.props.onChange(event.target.value);
    }

    render() {
        const element = this.props.override
            ? <div>
                <label className="form__label" htmlFor={this.props.overrideId}>{this.props.overrideLabel}</label>
                <input
                    className="input"
                    id={this.props.overrideId}
                    type="text"
                    onChange={this.handleChange}/>
            </div>
            :
            <div>
                <label className="form__label" htmlFor={this.props.id}>{this.props.label}</label>
                <select
                    className="select"
                    id={this.props.id}
                    onChange={this.handleChange}>
                    <option>Select</option>
                    {this.props.contents.map((list, index) => {
                        return <option key={index}>{list}</option>
                    })}
                </select>
            </div>;

        return (
            <div className="margin-bottom--2">
                {element}
            </div>
        )
    }
}

Select.propTypes = propTypes;
export default Select;
