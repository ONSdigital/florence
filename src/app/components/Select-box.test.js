import React from "react";
import Select from "./Select-box.jsx";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

const contentsList = ["a", "b", "c"];

test("Renders a select element when override is false", () => {
    const props = {
        override: false,
        contents: contentsList
    };
    const component = renderer.create(<Select {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
});

test("Renders an input element when override is true", () => {
    const props = {
        override: true
    };
    const component = renderer.create(<Select {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
});
