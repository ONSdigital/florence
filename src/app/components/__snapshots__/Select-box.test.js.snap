// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a select element when override is false 1`] = `
<div
  className="margin-bottom--2"
>
  <div>
    <label
      className="form__label"
    />
    <select
      className="select"
      onChange={[Function]}
    >
      <option>
        Select
      </option>
      <option>
        a
      </option>
      <option>
        b
      </option>
      <option>
        c
      </option>
    </select>
  </div>
</div>
`;

exports[`Renders an input element when override is true 1`] = `
<div
  className="margin-bottom--2"
>
  <div>
    <label
      className="form__label"
    />
    <input
      className="input"
      onChange={[Function]}
      type="text"
    />
  </div>
</div>
`;
